src/capi.cpp

```diff
@@ -276,6 +357,15 @@ void rtcInitLogger(rtcLogLevel level, rtcLogCallbackFunc cb) {
 
 void rtcSetUserPointer(int i, void *ptr) { setUserPointer(i, ptr); }
 
+void * rtcGetUserPointer(int i) {
+    auto userPointer = getUserPointer(i);
+    if (userPointer.has_value()) {
+        return userPointer.value();
+    } else {
+        return NULL;
+    }
```

Comment:
Just a note: this should be shortened to `return getUserPointer(i).value_or(nullptr);`